test/TritonGPU/coalesce.mlir

```diff
@@ -118,3 +118,40 @@ tt.func public @load_tensors_two_types(%arg0: !tt.ptr<f32> {tt.divisibility = 16
 }
 
 }
+
+// -----
+
+#mma = #triton_gpu.nvidia_mma<{warpsPerCTA = [8, 1], CTAsPerCGA = [1, 1], CTASplitNum = [1, 1], CTAOrder = [1, 0], instrShape = [16, 256, 16]}>
+
+// CHECK-LABEL: @fwd_kernel
+module attributes {"triton_gpu.num-warps" = 4 : i32, "triton_gpu.threads-per-warp" = 16 : i32} {
+  tt.func public @fwd_kernel(%arg0: !tt.ptr<f16> , %arg1: !tt.ptr<f16> , %arg2: !tt.ptr<f16> , %arg3: f32, %arg4: !tt.ptr<f32> , %arg5: !tt.ptr<f16> , %arg6: i32 , %arg7: i32 , %arg8: i32 , %arg9: i32 , %arg10: i32 , %arg11: i32 , %arg12: i32 , %arg13: i32 , %arg14: i32 , %arg15: i32 , %arg16: i32 , %arg17: i32 , %arg18: i32, %arg19: i32, %arg20: i32 , %arg21: i32) {
+    %cst = arith.constant dense<0.000000e+00> : tensor<128x16xf32, #mma> 
+    %cst_0 = arith.constant dense<0.000000e+00> : tensor<128xf32, #triton_gpu.slice<{dim = 1, parent = #mma}>> 
```

Comment:
can you minimize the test?

Reply:
The test was reduced (minimized) from a much larger test. I might be able to reduce it a bit more (remove some arguments). Let me try.

Reply:
> The test was reduced (minimized) from a much larger test.

If we accept patches with large testcases like this, our codebase quickly becomes difficult to maintain.  Indeed I've lost many days dealing with failures in tests like this.

You may need to write a testcase by hand.  Coming up with clear, small testcases is a critical part of contributing to Triton.

Reply:
Done. Now the test contains just a couple of  operations.